pkg/operator/k8sutil/network_test.go

```diff
@@ -26,77 +25,128 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestNetwork_ApplyMultusShort(t *testing.T) {
-	net := cephv1.NetworkSpec{
-		Provider: "multus",
-		Selectors: map[string]string{
-			publicNetworkSelectorKeyName:  "macvlan@net1",
-			clusterNetworkSelectorKeyName: "macvlan@net2",
-		},
-	}
-
-	tests := []struct {
-		name   string
-		labels map[string]string
-		want   []string
-	}{
-		{
-			name: "for a non osd pod",
-			want: []string{"macvlan@net1"},
-		},
-		{
-			name:   "for an osd pod",
-			labels: map[string]string{"app": "rook-ceph-osd"},
-			want:   []string{"macvlan@net1", "macvlan@net2"},
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
+func TestApplyMultus(t *testing.T) {
+	t.Run("short format", func(t *testing.T) {
+		tests := []struct {
+			name         string
+			netSelectors map[string]string
+			labels       map[string]string
+			want         string
+		}{
+			{
+				name: "for a non-osd pod",
+				netSelectors: map[string]string{
+					publicNetworkSelectorKeyName:  "macvlan@net1",
+					clusterNetworkSelectorKeyName: "macvlan@net2",
+				},
+				want: "macvlan@net1",
+			},
+			{
+				name: "for an osd pod",
+				netSelectors: map[string]string{
+					publicNetworkSelectorKeyName:  "macvlan@net1",
+					clusterNetworkSelectorKeyName: "macvlan@net2",
+				},
+				labels: map[string]string{"app": "rook-ceph-osd"},
+				want:   "macvlan@net1, macvlan@net2",
+			},
+			{
+				name: "for an osd pod (reverse ordering)",
+				netSelectors: map[string]string{
+					publicNetworkSelectorKeyName:  "macvlan@net2",
+					clusterNetworkSelectorKeyName: "macvlan@net1",
+				},
+				labels: map[string]string{"app": "rook-ceph-osd"},
+				want:   "macvlan@net1, macvlan@net2", // should not change the order of output
+			},
+		}
+		for _, test := range tests {
+			t.Run(test.name, func(t *testing.T) {
+				net := cephv1.NetworkSpec{
+					Provider:  "multus",
+					Selectors: test.netSelectors,
+				}
+				objMeta := metav1.ObjectMeta{}
+				objMeta.Labels = test.labels
+				err := ApplyMultus(net, &objMeta)
+				assert.NoError(t, err)
+				assert.Equal(t, test.want, objMeta.Annotations["k8s.v1.cni.cncf.io/networks"])
+			})
+		}
+	})
+
+	t.Run("JSON format", func(t *testing.T) {
+		json1 := `{"name": "macvlan", "interface": "net1"}`
+		json2 := `{"name": "macvlan", "interface": "net2"}`
+
+		t.Run("for a non-osd pod", func(t *testing.T) {
+			net := cephv1.NetworkSpec{
+				Provider: "multus",
+				Selectors: map[string]string{
+					"server": json1,
+					"broker": json2,
+				},
+			}
 			objMeta := metav1.ObjectMeta{}
-			objMeta.Labels = test.labels
 			err := ApplyMultus(net, &objMeta)
 			assert.NoError(t, err)
-			appliedNetworksList := strings.Split(objMeta.Annotations["k8s.v1.cni.cncf.io/networks"], ", ")
-			assert.ElementsMatch(t, appliedNetworksList, test.want)
+			// non-osd pods should not get any network annotations
+			// TODO: is this correct?
```

Comment:
@leseb @rohan47

Reply:
non-osd pods should not get cluster network annotations